validation/markets-validator.js

```diff
@@ -464,7 +481,48 @@ function MetadataValidator(metadata) {
             this.errors.push(new NullAttributeIssue("url"));
         } else if (url.length === 0) {
             this.errors.push(new EmptyAttributeIssue("url"));
-        }
+        } else {
+            this.validateUrlStatus(url);
+        }
+    };
+
+    this.validateUrlStatus = function (url) {
+        url = modurl.parse(url);
+        url.method = 'HEAD';
+        url.timeout = 10000; // 10 seconds
+
+        var request;
+        switch(url.protocol) {
+            case 'http:': request = http; break;
+            case 'https:': request = https; break;
+            default:
+                this.warnings.push(new CustomIssue(
+                    'Unknown protocol "' + url.protocol + '" in data source url.'
+                ));
+                return;
+        }
+
+        request = request.request(url, this.validateUrlStatusCallback);
+        request.on('error', function (error) {
+            asyncWarnings.push(new CustomIssue(
+                cityName + ': Error while accessing data source url: ' + error
+            ));
+        });
+        request.end();
+    };
+
+    this.validateUrlStatusCallback = function (response) {
```

Comment:
- Please remove space here `function (response)` for consistency.
- Please use `{ ... }` after an `if` statement.
- Please use a dedicated _issue_ such as:

``` js
function HttpResponseStatusIssue(cityName, statusCode) {

    this.cityName = cityName;
    this.statusCode = statusCode;

    this.toString = function() {
        return this.cityName + ": HTTP response status of data source url was: " + statusCode;
    };
}
```